src/client/services/conversation/conversation.js

```diff
@@ -89,6 +92,46 @@ var ConversationService = SparkBase.extend(
     this.encryptionDisabled = false;
   },
 
+  getExifData: function getExifData(file, buf) {
+    return new Promise(function(resolve) {
+      if (file && file.image && file.mimeType === 'image/jpeg') {
+        new ExifImage({ image : buf }, function (error, exifData) {
+          if (error) {
+            this.logger.info('conversation: error with image rotation ', error.message);
+          }
+          else {
+            if (exifData) {
+              file.image.orientation = exifData.image.Orientation;
+            }
+          }
+          resolve(buf);
+        });
+      }
+      else {
+        resolve(buf);
+      }
+    }.bind(this))
+  },
+
+  fixImageOrientation: function fixImageOrientation(file) {
```

Comment:
hmm, it still feels really to hang a method off the conversation service that does image fixing. Why does this method need to be invoked explicitly? Doesn't ShareActivity#add (ShareActivity#addFile in the legacy sdk) add create the thumbnail? Is there a reason why we can't rotate the thumbnail when we create it?